packages/playwright-test/src/expect.ts

```diff
@@ -248,10 +250,11 @@ class ExpectMetaInfoProxyHandler {
   }
 }
 
-async function pollMatcher(matcherName: any, isNot: boolean, timeout: number, generator: () => any, ...args: any[]) {
+async function pollMatcher(matcherName: any, isNot: boolean, pollIntervals: number[] | undefined, timeout: number, generator: () => any, ...args: any[]) {
   let matcherError;
   const startTime = monotonicTime();
-  const pollIntervals = [100, 250, 500];
+  pollIntervals = pollIntervals || [100, 250, 500, 1000];
+  const lastPollInterval = pollIntervals[pollIntervals.length - 1] || 1000;
```

Comment:
I assume if someone passes [0] as poll intervals we want to respect it
```suggestion
  const lastPollInterval = pollIntervals[pollIntervals.length - 1] ?? 1000;
```

Reply:
I'm not sure we ever want to see 0 here? That'd be a denial of service more or less...